libfreefare/freefare_internal.h

```diff
@@ -141,6 +141,8 @@ void		*assert_crypto_buffer_size(FreefareTag tag, size_t nbytes);
 #define MIFARE_ULTRALIGHT_C_PAGE_COUNT_READ 0x2C
 // Max PAGE_COUNT of the Ultralight Family:
 #define MIFARE_ULTRALIGHT_MAX_PAGE_COUNT 0x30
+// Default timeout (ms) for tag operations
+#define MIFARE_DEFAULT_TIMEOUT 500
```

Comment:
Not sure about this default value since your initial problem is that some operations need more than 500 ms.

Maybe it makes sense to raise it to 1000 ms, or maybe default to no timeout as before but only provide this API to allow end-users add timeout if they think it's useful for them?

I do not really care about what is chosen here (longer default timeout or no timeout by default), but the man page needs to be adjusted accordingly.

Reply:
I don't know why 500ms was chosen. It must have been for a reason and it got eventually approved and merged into HEAD

I had severe problems with that value which led me to implement the new function.

For the meantime I have not changed that "magic" default value.
I will be doing some tests and propose a new value which does not cause problems - at least in my use-case.

Reply:
Since there are no timeout [until recently](https://github.com/nfc-tools/libfreefare/commit/2d9ada743b79cee2ef8d466001592e18c53145f7), I guess defaulting to 0 is fine.

ping @manu0401

Reply:
If everything works with a 800ms timeout, then I guess that a 1s timeout by default is acceptable.  Or maybe 2 or 3 seconds to be absolutely sure?

Reply:
Fixed, timeout is now 2 seconds. This allows all long operations (like format) to succeed